src/server/routes/adminLogs.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2023 Shivam Awasthi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as propHelpers from '../../client/helpers/props';
+import {escapeProps, generateProps} from '../helpers/props';
+import AdminLogsPage from '../../client/components/pages/adminLogs';
+import Layout from '../../client/containers/layout';
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import express from 'express';
+import {getNextEnabledAndResultsArray} from '../../common/helpers/utils';
+import {getOrderedAdminLogs} from '../helpers/adminLogs';
+import {parseQuery} from '../helpers/utils';
+import target from '../templates/target';
+
+
+const router = express.Router();
+
+router.get('/', async (req, res, next) => {
+	const {orm} = req.app.locals;
+	const size = req.query.size ? parseInt(parseQuery(req.url).get('size'), 10) : 20;
+	const from = req.query.from ? parseInt(parseQuery(req.url).get('from'), 10) : 0;
```

Comment:
```suggestion
	const query = parseQuery(req.url);
	const size = parseInt(query.get('size'), 10) || 20;
	const from = parseInt(query.get('from'), 10) || 0;
```
There's no need to do the URL parsing twice. Since there are also a few occurrences of this pattern in the code from #1000 you may want to also fix those while you're at it :innocent:
The other simplification works because `parseInt(null) === NaN` and `parseInt('') === NaN` and NaN is a falsy value. Considering how often this pattern is used maybe it is also worth having a helper function `getInt(query, fallback = 0)`?

Reply:
Thanks for the suggestions! I have made the changes.

Reply:
There are some other files where we can use these helper functions, but I guess its better to make those changes in a different PR later.

Reply:
Thank you also for finding time to lend an extra eye on these PRs.

Reply:
> Thank you also for finding time to lend an extra eye on these PRs.

I second that !